src/github.com/matrix-org/dendrite/common/log.go

```diff
@@ -15,26 +15,62 @@
 package common
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
-	"github.com/Sirupsen/logrus"
 	"github.com/matrix-org/dugong"
+	"github.com/mgutz/ansi"
+	"github.com/sirupsen/logrus"
 )
 
-type utcFormatter struct {
-	logrus.Formatter
+type dendriteFormatter struct {
+	logrus.TextFormatter
 }
 
-func (f utcFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+func (f dendriteFormatter) Format(entry *logrus.Entry) (format []byte, err error) {
 	entry.Time = entry.Time.UTC()
-	return f.Formatter.Format(entry)
+
+	if _, ok := entry.Data["prefix"]; ok {
+		prefix, ok := entry.Data["prefix"].(string)
+		if !ok {
+			return f.TextFormatter.Format(entry)
+		}
```

Comment:
Yep, re-reading these lines makes me think I messed this up (might have been to tired to think correctly yesterday)